.github/workflows/release.yml

```diff
@@ -86,3 +112,40 @@ jobs:
           MICRONAUT_BUILD_EMAIL: ${{ secrets.MICRONAUT_BUILD_EMAIL }}
         with:
           token: ${{ secrets.GITHUB_TOKEN }}
+
+  provenance:
+    needs: [release]
+    permissions:
+      actions: read # To read the workflow path.
+      id-token: write # To sign the provenance.
+      contents: write # To add assets to a release.
+    uses: slsa-framework/slsa-github-generator/.github/workflows/generator_generic_slsa3.yml@v1.2.0
+    with:
+      base64-subjects: "${{ needs.release.outputs.hashes }}"
+      upload-assets: true # Upload to a new release
+
+  github_release:
+    needs: [release, provenance]
+    runs-on: ubuntu-latest
+    if: startsWith(github.ref, 'refs/tags/')
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
+      - name: Download artifacts
+        uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
+        with:
+          name: gradle-build-outputs
+          path: dist
+      - name: Upload assets
+        # Upload the artifacts and SLSA L3 provenance as assets to the existing release.
+        # The artifacts for each module are stored separately in a zip file. Note that
+        # the provenance will attest to the artifact files and not the aggregated zip file.
+        run: |
+          ARTIFACTS=""
+          for dir in dist/* ; do
```

Comment:
I seems a bit strange to look for the artifacts in `build/repo`, but then use those in `dist` for uploading. I think we should use `build/repo` for both. Also I agree with @alvarosanchez that it would be nice if we had a single zip, with possibly nested zips in it.

If it helps we could even have a Gradle task to do it.

Reply:
> I think we should use build/repo for both.

Sure, I'll change that.